File: algorithms/2_medium/M/02216_minimum_deletions_to_make_array_beautiful/solution.c
int minDeletion(const int *restrict nums, const int len) {
    if (len <= 1) { return len; }

    int cnt = 1;
    int lo = 0;

    for (int hi = 1; hi < len; hi += 1) {
        if (((cnt - 1) & 1) == 0) {
            if (nums[lo] != nums[hi]) {
                lo = hi;
                cnt += 1;
            }
        } else {
            lo = hi;
            cnt += 1;
        }
    }

    if ((cnt & 1) == 1) {
        cnt -= 1;
    }

    const int needs = len - cnt;
    return needs;
}

File: algorithms/2_medium/M/02216_minimum_deletions_to_make_array_beautiful/solution.rs
impl Solution {
    pub fn min_deletion(nums: Vec<i32>) -> i32 {
        if nums.len() <= 1 { return nums.len() as i32; }

        let len = nums.len();
        let mut cnt: usize = 1;
        let mut lo: usize = 0;
        let mut hi: usize = 1;

        for hi in 1..len {
            if ((cnt - 1) & 1) == 0 {
                if nums[lo] != nums[hi] {
                    lo = hi;
                    cnt += 1;
                }
            } else {
                lo = hi;
                cnt += 1;
            }
        }

        if (cnt & 1) == 1 {
            cnt -= 1;
        }

        (len - cnt).try_into().unwrap()
    }
}
